source/shared/cpp/ObjectModel/TextSection.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <string>
+#include <vector>
+#include "Enums.h"
+
+namespace AdaptiveCards
+{
+    class TextSection
+    {
+    public:
+        TextSection();
+        TextSection(std::string text, TextSectionFormat format);
+        TextSection(std::string text, std::string originalText, TextSectionFormat format);
+
+        std::string GetText() const;
+        std::string GetOriginalText() const;
+        TextSectionFormat GetFormat() const;
+        // Date format should be: MM/DD/YYYY
+        int GetDay() const;
```

Comment:
why do we need day/month/year in integer?

Reply:
it's easier to retrieve them if the format is not as expected, also, in the case of c++ the date formatting needs a tm struct that needs to be filled by hand so having this deal with the day month year splitting is easier

Reply:
if uwp doesn't have a date and time formatter, you can consider saving tm struct saved from textblocktext.


---
In reply to: [168269415](https://github.com/Microsoft/AdaptiveCards/pull/1102#discussion_r168269415) [](ancestors = 168269415)

Reply:
the only date time formatter I found for uwp was the COleDateTime but what I get from the format functions is that it's either language or short and long format

 [https://docs.microsoft.com/en-us/cpp/atl-mfc-shared/reference/coledatetime-class#format](https://docs.microsoft.com/en-us/cpp/atl-mfc-shared/reference/coledatetime-class#format)

I think I'll end up saving the tm struct in the TextSection class then

---
In reply to: [168272931](https://github.com/Microsoft/AdaptiveCards/pull/1102#discussion_r168272931) [](ancestors = 168272931,168269415)